### PR TITLE
Heat map cancelTimeout, current heatmap toggle btn

### LINKED ISSUE #33: heatmap setTimeout problems
There is a problem with current heatmap and 24hours heatmap. Both heatmaps are using setTimeout to show another heatmap step or return to marker view. Unfortunately these callbacks are not canceled while clicking other button. 

Problem with current heatmap should be fixed according to these comments [1](https://github.com/Nuanda/smogmapper/pull/25#issuecomment-169651744), [2](https://github.com/Nuanda/smogmapper/pull/25#issuecomment-169653708). For 24h heatmap setTimeout should be canceled after clicking other action button.

### PR DESCRIPTION
**Pull request by mkasztelnik**
After clicking 24h heat map or current heat map button active timeout is canceled. What is more current heat map button is changed into toggle button.

Fixes #33

### CODE REVIEW

File: app/assets/javascripts/smog_map.js.coffee [hunk 1]
```diff
@@ -67,3 +67,28 @@ class @SmogMap
...
+      window.smogMap.addLayer(sensorMarker)
+
+  hideSensors: ->
+    window.markerLayer.eachLayer (sensorMarker) ->
+      window.smogMap.removeLayer(sensorMarker)
+
+  showHeatmap: ->
+   unless window.heatmapLayer
+         window.heatmapLayer = new HeatmapOverlay(heatmapConfig)
+    window.smogMap.addLayer(window.heatmapLayer)
```
Comment (Nuanda) [inline app/assets/javascripts/smog_map.js.coffee]: That's a strange indent :).

Comment (mkasztelnik) [inline app/assets/javascripts/smog_map.js.coffee]: fixed

**Comment by Nuanda** [comment]
1. Are you able to 'reset' the 24h animation when the user clicks it another time when it runs? Or perhaps simply block (disable) the button while it runs? I'm not sure if that's intentional but it sometimes behaves like it resets the animation, but at other times it does not, and two animations run one by the other? Maybe a race condition in the js? If so, perhaps disabling the button is a safer way out.

2. Have you tested the new buttons on a mobile?

**Comment by Nuanda** [comment]
3. Also, please turn on markers when one hits the 'current situation' button and the markers are invisible (during the animation).

**Comment by mkasztelnik** [comment]
Ad 1. there is no race condition. The problem is connected with following scenario: heatmap values are set into heatmap layer (from previous 24h heatmap run or current heatmap). While starting new 24h heatmap we are showing heatmap at the beginning, next ajax query is made, that is why blink at the beginning. 

Ad 2. Not yet, I will do this when 2 and 3 will be fixed.

**Comment by mkasztelnik** [comment]
1 and 3 fixed. Tested on nexus 4 (unfortunately heatmap is not shown using this branch and `master` - grey layout color, nothing else :() and nexus 7 (I needed to close all application besides browser to make it work - but it works then as expected). 

@Nuanda please review it once again

**Pull request merged**